### DISCUSSION TITLE
Feature to enable profiling the application

### DISCUSSION THREAD

**Discussion by WoLfulus** [directus/directus#3027] [Feature Requests]
It would be nice to have a way to do some profiling on the application (and expose it to be used by extensions too).

If we implement this as events, we open the possibility to have extensions that does monitoring/reporting to external services too.

- Example events would be reporting database queries and the time spent. 
- The time spent in each route, in a custom route, extensions, hooks, etc.
- Core parts of Directus

**Comment by rijkvanzanten** [comment]
Lets start with:

* DB Queries
* Service methods (eg `FieldsService#readAll`)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
